source/portable/NetworkInterface/DriverSAM/NetworkInterface.c

```diff
@@ -680,9 +686,18 @@ static BaseType_t prvGMACInit( NetworkInterface_t * pxInterface )
     NVIC_SetPriority( GMAC_IRQn, configMAC_INTERRUPT_PRIORITY );
     NVIC_EnableIRQ( GMAC_IRQn );
 
+    /* Clear the hash table for multicast MAC addresses. */
+    GMAC->GMAC_HRB = 0U; /* Hash Register Bottom. */
+    GMAC->GMAC_HRT = 0U; /* Hash Register Top. */
+
+    /* gmac_enable_multicast_hash() sets the wrong bit, don't use it. */
+    /* gmac_enable_multicast_hash( GMAC, pdTRUE ); */
+    /* set Multicast Hash Enable. */
+    GMAC->GMAC_NCFGR |= GMAC_NCFGR_MTIHEN;
+
     #if ( ipconfigUSE_LLMNR == 1 )
         {
-            gmac_set_address( GMAC, xEntry++, xLLMNR_MacAdress.ucBytes );
+            prvAddMACAddress( xLLMNR_MacAdress.ucBytes );
```

Comment:
Extra check is needed that ipconfigUSE_IPv6  == 0.

Reply:
We shouldn't check `ipconfigUSE_IPv6 == 0` here. This mac address is used for IPv4 LLMNR, which is independent with IPv6.